1.0 BlogPost (#4402)
knative/docs: blog/docs/articles/knative-1.0.md

---
title: "Knative 1.0: What's new?"
linkTitle: "Knative 1.0: What's new?"
date: 2021-11-02
description: "Details on the 1.0 release of Knative"
type: "blog"
---

![1.0 Banner Image](/blog/images/1.0Banner.jpg)

***Authors: Carlos Santana (IBM), Omer Bensaadon (VMware), Maria Cruz (Google)***

Today we release Knative 1.0, reaching an important milestone made possible thanks to the contributions and collaboration of over 600 developers. The Knative project was released by Google in July 2018, and it was developed in close partnership with VMWare, IBM, Red Hat, and SAP. Over the last 3 years, Knative overcame many limitations to become [the most widely-installed serverless layer on Kubernetes.](https://www.cncf.io/wp-content/uploads/2020/11/CNCF_Survey_Report_2020.pdf)

## What's New in 1.0
In addition to a lot of bug fixes, stability and performance enhancements, the community has shipped the following improvements (in approximately chronological order):

> * Support for multiple HTTP routing layers beyond Istio (Contour, Kourier, Ambassador have all implemented this layer)

Author: Cheeky Istio first ;)

> * Support for multiple HTTP routing layers beyond Istio (Contour, Kourier, Ambassador have all implemented this layer)

Author: I think we got rid of Gloo

> * Support for multiple HTTP routing layers beyond Istio (Contour, Kourier, Ambassador have all implemented this layer)

Author: Or rather, there were lots of tests Gloo wasn't passing

* Eventing supports customizing delivery options for event destinations, including retries and dead-letter queues for undeliverable messages
* Event tracing support for both Brokers and Channels for improved debugging
* Spawned the Tekton project from Knative Build
* Added Parallel and Sequence components to codify certain composite eventing workflows.
* Documented and cataloged Event Sources and how to contribute them (currently covering about 40 different event sources)
* Tested “hitless” upgrades with no dropped requests between minor version releases
* Redesigned API shapes for Serving to match PodTemplateSpec used by Deployment, CronJob, etc to simplify usage for Kubernetes-aware users
* Added support for injecting event destination addresses into PodTemplateSpec shaped objects
* Support for HPA autoscaling as well as Knative Pod Autoscaling based on concurrency or rps.
* High availability of control plane components using leader election slices
* Added an operator to help administrators install Knative
* Added a quickstart for developer try Knative locally
* Simplified management and publication of Services using DomainMapping (beta)

## Get Involved!

> Join the Knative community meetup on November 17, 2021, to hear from Knative maintainer Ville Aikas about the latest changes coming with Knative 1.0. Newcomers to the Knative community are always welcome! Join the Knative Slack space to ask questions and troubleshoot issues as you get acquainted with the project. Finally, find all the project documentation on the Knative website and contribute to the project on GitHub.

Author: Oooo love the plug

* We still want to keep the versions in sync and decided to go for 1.0 for all components.
* GA levels are marked for components individually.

??? question "Why number all components 1.0?""
    Two reasons, one user-facing and one contributor-facing. The big user-facing reason is that it gives users a single number to hang on to when understanding what they've installed and what works together. The smaller contributor-facing reason is that all our infrastructure is designed to manage a single version number, and updating it to support multiple version numbers doesn't seem like a good use of time given the first point.

??? question "Isn't "1.0 and Beta" for a component like eventing-rabbitmq or eventing-kafka-broker confusing?""
    Unless we wait for everything related to Knative to be done, we'll always have some components or features that are in an alpha or beta state. While this sometimes happens along a component boundary, it can also happen within a component, so the version number can't be a sole indicator of "GA or not". (This happens to other projects like Kubernetes as well, and to specific features in Serving or Eventing.)

>     Going forward, one of the things we'll need to be clear about is the maturity level of various components or features, and moving features along the path to either GA or retirement.

Reviewer: @omerbensaadon @csantanapr +1 on removing this part.

>     Going forward, one of the things we'll need to be clear about is the maturity level of various components or features, and moving features along the path to either GA or retirement.

Author: We'd like to keep it 😅 

There are a lot of folks that rely on Knative so having this section is important to us

> ---
> 
> **Only read past this point if you're curious about the reasoning about the release mechanics; there's no actionable information below.**
> 
> * Knative is composed of many components that are versioned together.
> * Those components have different levels of maturity, ranging from already-GA to experimental.
> * We still want to keep the versions in sync and decided to go for 1.0 for all components.
> * GA levels are marked for components individually.
> 
> ??? question "Why number all components 1.0?""
>     Two reasons, one user-facing and one contributor-facing. The big user-facing reason is that it gives users a single number to hang on to when understanding what they've installed and what works together. The smaller contributor-facing reason is that all our infrastructure is designed to manage a single version number, and updating it to support multiple version numbers doesn't seem like a good use of time given the first point.
> 
> ??? question "Isn't "1.0 and Beta" for a component like eventing-rabbitmq or eventing-kafka-broker confusing?""
>     Unless we wait for everything related to Knative to be done, we'll always have some components or features that are in an alpha or beta state. While this sometimes happens along a component boundary, it can also happen within a component, so the version number can't be a sole indicator of "GA or not". (This happens to other projects like Kubernetes as well, and to specific features in Serving or Eventing.)
> 
>     Going forward, one of the things we'll need to be clear about is the maturity level of various components or features, and moving features along the path to either GA or retirement.

Reviewer 2: ```suggestion
```

>     Going forward, one of the things we'll need to be clear about is the maturity level of various components or features, and moving features along the path to either GA or retirement.

Reviewer 2: moved to its own section; I actually think it's very relevant in that context.

---
title: "Knative 1.0 is out!"
linkTitle: "Knative 1.0 is out!"
date: 2021-11-02
description: "Details on the 1.0 release of Knative"
type: "blog"
---

![1.0 Banner Image](/blog/images/1.0Banner.jpg)

> ***Authors: Carlos Santana (IBM), Omer Bensaadon (VMware), Maria Cruz (Google)***

Reviewer 3: ```suggestion
***Authors: [Carlos Santana](https://twitter.com/csantanapr) (IBM), [Omer Bensaadon](https://twitter.com/omer_bensaadon) (VMware), [Maria Cruz](https://twitter.com/marianarra_) (Google)***
```

link hanldes

Reviewer 2: Feel free to consider if my contributions are worth a byline.

## Learn more
Knative steering committee member Ville Aikas is [the guest on the Kubernetes Podcast from Google this week](https://kubernetespodcast.com/episode/166-knative-1.0/), telling the story of the creation of the project and its journey to 1.0.  You can also join [the Knative community meetup on November 17,](https://calendar.google.com/calendar/u/0/r/eventedit/NnAycjJyZmdlMTF1b2FuOGJzZjZ1dXA0aTZfMjAyMTExMjRUMTczMDAwWiBrbmF0aXZlLnRlYW1fOXE4M2JnMDdxczViOXJyc2xwNWpvcjRsNnNAZw?tab=mc) where Ville will talk about the latest changes to the project.

## Get involved
Newcomers to the Knative community are always welcome! [Join the Knative Slack space](https://slack.knative.dev/) to ask questions and troubleshoot issues as you get acquainted with the project. Finally, find all the [project documentation](https://knative.dev/docs/) on the Knative website and [contribute to the project on GitHub](https://github.com/knative).

## Thank you to our contributors!
Achieving this milestone has truly been a community effort — we’d be remiss not to thank some of the people who helped us get here.
Thanks to…

> * Knative Supporting Companies
>     * Google
>     * IBM
>     * Red Hat
>     * SAP
>     * TriggerMesh
>     * VMware
>     * [And many more](https://knative.teststats.cncf.io/d/5/companies-table?orgId=1&var-period_name=Last%202%20years&var-metric=contributions)
> * The Knative Technical Oversight Committee
> * The Knative Steering Committee
> * The Knative Trademark Committee
> * All Knative Contributors, past and present
> * Google for sponsoring our website, testing infrastructure and producing the community meetup every month.

Reviewer 2: ```suggestion
* The companies supporting Knative, including
    * Google (who also sponsor our website and testing infrastructure, and produce the community meetup every month)
    * IBM
    * Red Hat
    * SAP
    * TriggerMesh
    * VMware
    * [and many more!](https://knative.teststats.cncf.io/d/5/companies-table?orgId=1&var-period_name=Last%20decade&var-metric=contributions)
* The members of our [Technical Oversight Committee](https://github.com/knative/community/blob/main/TECH-OVERSIGHT-COMMITTEE.md), [Steering Committee](https://github.com/knative/community/blob/main/STEERING-COMMITTEE.md) and [Trademark Committee](https://github.com/knative/community/blob/main/TRADEMARK-COMMITTEE.md)
* All Knative contributors, past and present
```

---
title: "Knative 1.0 is out!"
linkTitle: "Knative 1.0 is out!"
date: 2021-11-02
description: "Details on the 1.0 release of Knative"
type: "blog"
---

![1.0 Banner Image](/blog/images/1.0Banner.jpg)

***Authors: [Carlos Santana](https://twitter.com/csantanapr) (IBM), [Omer Bensaadon](https://twitter.com/omer_bensaadon) (VMware), [Maria Cruz](https://twitter.com/marianarra_) (Google)***

Today we release Knative 1.0, reaching an important milestone made possible thanks to the contributions and collaboration of over 600 developers. The Knative project was released by Google in July 2018, and it was developed in close partnership with VMWare, IBM, Red Hat, and SAP. Over the last 3 years, Knative has become [the most widely-installed serverless layer on Kubernetes.](https://www.cncf.io/wp-content/uploads/2020/11/CNCF_Survey_Report_2020.pdf)

## What's new

In case you haven't been following Knative's development closely, there have been a lot of changes since our initial release in July 2018.

In addition to a myriad of bug fixes, stability and performance enhancements, our community has shipped the following improvements, in approximately chronological order:

> * Support for multiple HTTP routing layers beyond (including Istio, Contour, Kourier and Ambassador)

Reviewer 2: ```suggestion
* Support for multiple HTTP routing layers (including Istio, Contour, Kourier and Ambassador)
```

> Cheeky Istio first ;)

I actually forgot to remove the most important word.  The point was, it Istio-only at launch, and others were added later.

* Support for automatic provisioning of TLS certificates (either via DNS or HTTP01 challenges)
* Custom delivery options for event destinations, including retries and dead-letter queues for undeliverable messages
* Event tracing support for both Brokers and Channels, for improved debugging
* [The Tekton project](https://tekton.dev/), spawned from Knative Build
* Parallel and Sequence components, to codify certain composite eventing workflows
* Documentation for Event Sources and how to contribute them, currently covering about 40 different event sources
* "Hitless” upgrades with no dropped requests between minor version releases
* Redesign of API shapes for Serving to match the PodTemplateSpec used by Deployment, CronJob, etc, to simplify usage for Kubernetes-aware users
* Support for injecting event destination addresses into PodTemplateSpec shaped objects
* Support for horizontal pod autoscaling based on concurrency or RPS
* High availability of control plane components using leader election slices
* An operator, to help administrators install Knative
* A quickstart, for developers to try Knative locally
* Simplified management and publication of Services using DomainMapping

> 

Reviewer 2: ```suggestion

## What does it mean to be 1.0?

Knative is composed of many components that are versioned together.  Those components have different levels of maturity, ranging from "experimental" to "already GA" (Generally Available).  We still want to keep the versions in sync, and so have decided to move all components to version 1.0.  GA levels are marked for components individually.

### Why move all the components to 1.0 at once?

Two reasons: one user-facing, and one contributor-facing. The big, user-facing reason is that it gives users a single number to hang on to when understanding what they've installed and what works together. The smaller, contributor-facing reason is that all our infrastructure is designed to manage a single version number, and updating it to support multiple version numbers doesn't seem like a good use of time given the first point.

### Isn't a component being both "1.0" and "Beta" confusing? 

Unless we wait for everything related to Knative to be done, we'll always have some components or features that are in an alpha or beta state. While this sometimes happens along a component boundary, it can also happen within a component, so the version number can't be a sole indicator of "GA or not". (This happens to other projects like Kubernetes as well, and to specific features in Serving or Eventing.)

Going forward, one of the things the project will be clear about is the maturity level of various components or features, and moving features along the path to either GA or retirement.

```

Knative steering committee member Ville Aikas is [the guest on the Kubernetes Podcast from Google this week](https://kubernetespodcast.com/episode/166-knative-1.0/), telling the story of the creation of the project and its journey to 1.0.  You can also join [the Knative community meetup on November 17,](https://calendar.google.com/calendar/u/0/r/eventedit/NnAycjJyZmdlMTF1b2FuOGJzZjZ1dXA0aTZfMjAyMTExMjRUMTczMDAwWiBrbmF0aXZlLnRlYW1fOXE4M2JnMDdxczViOXJyc2xwNWpvcjRsNnNAZw?tab=mc) where Ville will talk about the latest changes to the project.

## Get involved
Newcomers to the Knative community are always welcome! [Join the Knative Slack space](https://slack.knative.dev/) to ask questions and troubleshoot issues as you get acquainted with the project. Finally, find all the [project documentation](https://knative.dev/docs/) on the Knative website and [contribute to the project on GitHub](https://github.com/knative).

## Thank you to our contributors!
Achieving this milestone has truly been a community effort — we’d be remiss not to thank some of the people who helped us get here.
Thanks to…

* Knative Supporting Companies
  * Google
  * IBM
  * Red Hat
  * SAP
  * TriggerMesh

>   * VMware

Reviewer 2: These aren't showing as indented properly (see https://deploy-preview-4402--knative.netlify.app/blog/articles/knative-1.0/)

Reviewer 2: You may also like to link to their respective Knative-powered services?
